File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        res = 0
        cheapest = prices[0]
        for i in range(1, len(prices), 1):
            res = max(res, prices[i]-cheapest)
            cheapest = min(cheapest, prices[i])
            
        return res

File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int n = heights.size(), lo = 0, hi = n-1;
        int res = 0;

        while(lo<hi){
            int ht = min(heights[lo], heights[hi]), len = hi-lo;
            res = max(res, ht * len);

            if(heights[lo] < heights[hi])
                lo++;
            else
                hi--;
        }

        return res;
    }
};
